src/tdvp_step.jl

```diff
@@ -192,14 +211,32 @@ function tdvp_site_update!(
   maxdim,
   mindim,
   maxtruncerr,
+  solver_which_eigenvalue,
+  solver_tol,
+  solver_krylovdim,
+  solver_maxiter,
+  solver_verbosity,
+  ishermitian,
 )
   N = length(psi)
   nsite = 1
   # Do 'forwards' evolution step
   set_nsite!(PH, nsite)
   position!(PH, psi, b)
   phi1 = psi[b]
-  phi1, info = solver(PH, time_step, phi1; current_time, outputlevel)
+  phi1, info = solver(
+    PH,
+    time_step,
+    phi1;
+    current_time,
+    outputlevel,
+    solver_which_eigenvalue,
```

Comment:
Aren't these particular to certain solvers? It seems like this would be a good candidate for passing a `NamedTuple` called `solver_kwargs`.

Reply:
Good feedback. This is something I still haven't grasped about what our best practices should be now. From the discussion with Ash I'm definitely convinced of the following two things:
- no more using `get` to grab keyword args
- lowest level of the function stack should take definite keyword args
- higher levels needing certain args along the way down should "pop" them by explicitly asking for them in their arguments

What I haven't thought through all the way is when and how best to use NamedTuples. Like in some of the solvers things are packaged into NamedTuples only to be splatted right back into the KrylovKit function. This makes sense for saving some typing but is that the only reason?

Reply:
By the way, I'm thinking of adding a new page to Documenter about this under a Development section called Keyword Arguments and we can put some "best practices" there for future reference. (My future reference as well as others contributing to ITensor.)

Reply:
I think there are two aspects:
1. Not all solvers will have those specific keyword arguments, and in fact we should be mostly agnostic about which keyword arguments solvers should accept (some generic keyword arguments like `outputlevel` may be passed to solvers, but we should be careful and clear about which ones will get passed to all solvers).
2. Organizing it into a NamedTuple creates an organization so that users can indicate a group of keyword arguments that will get passed to the solver.

Reply:
Thanks - agreed. Will work on this more.

Reply:
I wanted to check also why we were using the NamedTuple approach inside the solvers. 

Is it to give us the ability to choose our own names for certain kwargs versus having to use KrylovKit's?  And I guess to make it a bit nicer looking code with args on different lines?

Reply:
I think it is mostly a style choice of organizing which keyword arguments get passed to which internal function, so for example I could imagine an interface choice between:
```julia
tdvp(H, t, psi; nsweeps=10, cutoff=1e-8, solver_kwargs=(; tol=1e-8, krylovdim=20, maxiter=10, verbosity=1, ishermitian=true))
```
vs.
```julia
tdvp(H, t, psi; nsweeps=10, cutoff=1e-8, tol=1e-8, krylovdim=20, maxiter=10, verbosity=1, ishermitian=true)
```
I like the first style because for the more generic keyword arguments like `tol` you might not know what it is referring to/where it will actually get used. Otherwise, we would have to come up with our own names for keyword arguments, but then that would require doing specific mappings for certain keyword arguments which isn't extensible to unknown solvers.

It would also help with code organization since we can just pass `solver_kwargs` down through the different levels of functions then splat it into the solver at the bottom level.

Reply:
That's helpful, and I see the reason now.

The main question, then, is whether this is the right PR to explore this since I don't think the other solvers accept a named tuple currently. So we'd have to change them all together so as not to break them, correct? I was wanting to do a larger kwargs cleanup / refactor here anyway (e.g. eliminate calls to `get` in the tdvp_step, tdvp_sweep functions) so what do you think about a whole kwarg oriented PR after this?

Reply:
Probably it is best to have a separate kwarg PR since that is a larger issue.

For the record, I wasn't proposing passing the `solver_kwargs` NamedTuple as a NamedTuple directly, but instead splatting it into the keyword arguments of the solver, for example:
```julia
solver(PH, time_step, phi1; current_time, outputlevel, solver_kwargs...)
```

Reply:
Ok great, we're in agreement about the PR for kwargs.

Thanks for the clarification about the kwargs. Then we should discuss further about the issue #27 where I commented just a minute ago about the idea of maybe passing the whole NamedTuple as a way around solvers having to either slurp on the one hand or all take the same set of kwargs on the other (unless I missed something there).

Lastly, there's a consideration about asking users to use NamedTuples. That's mostly a matter of documentation I think, though, where we just make sure there are clear enough doc strings and examples.